WcaOnRails/spec/controllers/competitions_controller_spec.rb

```diff
@@ -1013,15 +1013,22 @@ def add_result(pos, name)
         assert_enqueued_jobs 4
       end
 
-      it "sends notifications of id claim possibility to newcomers" do
+      it "assigns wca id when user matches results or sends notifications of id claim possibility to newcomers when not matched" do
```

Comment:
It would be best to split this into two separate test cases, so that each validates one scenario =)

Reply:
I did consider that, however I think there's also value in testing it mixed, so we can verify the two are not interfering with one another? I don't feel strongly about this, though. @viroulep a friendly tie breaker? :)

Reply:
Hum to be honest I don't feel strongly either way...
However I'd be happy to see a test where assigning the ID fails because there are more than one matches!
In that sense it may be simpler to keep the existing test for newcomers, and write a new one that deals with exactly one match or more than one.

Reply:
Done! Also added a test to make sure we're not assigning a WCA ID that's already assigned (although that should fail validation, but still). Please re-review the tests.